Code/Framework/AzCore/AzCore/std/string/string.h

```diff
@@ -1753,22 +1612,173 @@ namespace AZStd
 
         inline void deallocate_memory(pointer data, size_type expandedSize, const false_type& /* !allocator::allow_memory_leaks */)
         {
-            if (m_capacity >= SSO_BUF_SIZE)
+            if (!m_storage.first().ShortStringOptimizationActive())
             {
-                size_type byteSize = (expandedSize == 0) ? (sizeof(node_type) * (m_capacity + 1)) : expandedSize;
-                m_allocator.deallocate(data, byteSize, alignment_of<node_type>::value);
+                size_type byteSize = (expandedSize == 0) ? (sizeof(node_type) * (m_storage.first().GetCapacity() + 1)) : expandedSize;
+                m_storage.second().deallocate(data, byteSize, alignof(node_type));
             }
         }
 
-        union //Storage
+        //! Assuming 64-bit for pointer and size_t size
+        //! The offset and sizes of each structure are marked below
+
+        //! dynamically allocated data
+        struct AllocatedStringData
+        {
+            AllocatedStringData()
+            {
+                m_capacity = 0;
+                m_ssoActive = false;
+            }
+            // bit offset: 0, bits: 64
+            pointer m_data{};
+
+            // bit offset: 64, bit: 64
+            size_type m_size{};
```

Comment:
My opinion is that string classes don't really need to suport 64-bit sizes.
It is hard to work with actual string data that are a couple of megs large.
A string of 4 billion + characters isn't pratical.

A binary like `AZStd::vector<uint8_t>` makes sense to support 64-bit sizes, since files greater than 4GiB might be read into a buffer, but for a text based class it supporting more than 32-bits isn't really useful